jobs/upload-brew.Jenkinsfile

```
@@ -0,0 +1,117 @@
+])
+
+def build_description = "[${params.VERSION}]"
+
+// runtime parameter always wins
+def cosa_img = params.COREOS_ASSEMBLER_IMAGE 
+cosa_img = cosa_img ?: pipeutils.get_cosa_img(pipecfg, params.STREAM) 
+
+// We don't need that much memory for downloading/uploading to brew, since
+// it will be mostly metadata
+def cosa_memory_request_mb = 5 * 1024 as Integer
+
+// Same here, we don't need that much 
+def ncpus = 4
+
+echo "Waiting for upload-brew lock"
+currentBuild.description = "${build_description} Waiting"
+def stream_info = pipecfg.streams[params.STREAM]
+
+lock(resource: "upload-brew") {
+    timeout(time: 240, unit: 'MINUTES') {
+    cosaPod(cpu: "${ncpus}",
+            memory: "${cosa_memory_request_mb}Mi",
+            image: cosa_img,
+            serviceAccount: "jenkins", 
+            secrets: ["brew-keytab", "brew-ca:ca.crt:/etc/pki/ca.crt",
+                      "koji-conf:koji.conf:/etc/koji.conf",
+                      "krb5-conf:krb5.conf:/etc/krb5.conf"]) {
+    try {
+        stage('Fetch Metadata') {
+            def ref = pipeutils.get_source_config_ref_for_stream(pipecfg, params.STREAM)
+            def variant = stream_info.variant ? "--variant ${stream_info.variant}" : ""
+            def s3_stream_dir = pipeutils.get_s3_streams_dir(pipecfg, params.STREAM)
+            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                cosa init --branch ${ref} ${variant} ${pipecfg.source_config.url}
+                cosa buildfetch --build=${params.VERSION} \
+                    --arch=all --url=s3://${s3_stream_dir}/builds \
+                    --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG} \
+                    --artifact=qemu \
+                    --file "coreos-assembler-config-git.json"
+            """)
+        }
+        stage('Brew Upload') {
+            for (arch in arches) { 
+                shwrap("""
+                    coreos-assembler koji-upload \
+                        upload --reserve-id \
+                        --keytab "/run/kubernetes/secrets/brew-keytab/brew.keytab" \
+                        --build ${version} \
+                        --retry-attempts 6 \
+                        --buildroot builds \
+                        --owner ${brew_principle} \
+                        --profile ${brew_profile} \
+                        --tag ${params.TAG} \
+                        --arch "$arch"
+                """)
+            }
+        }
+        currentBuild.result = 'SUCCESS'
+
+    // main timeout and try {} finish here
+    } catch (e) {
+             currentBuild.result = 'FAILURE'
+             throw e
+         } finally {
+             def color
+             def message = "[${params.STREAM}][${params.VERSION}][${params.ARCH}]"
+
+             if (currentBuild.result == 'SUCCESS') {
+                 message = "brew :sparkles: ${message}"
+                 color = 'good';
+             } else {
+                 message = "brew :fire: ${message}"
```

Bikeshed:

```suggestion
                 message = ":beer: :sparkles: ${message}"
```

(Matches the Brew favicon) :)